composer/datasets/streaming/dataset.py

```diff
@@ -227,8 +234,9 @@ def download(self) -> None:
             self._lock = Lock()
 
         with self._lock:
-            if self._is_downloaded:
+            if self._download_status != DownloadStatus.NOT_STARTED:
```

Comment:
Would it make sense to call download a second time (i.e. if it failed) If so, then should this be check  to see if download_steatus != finished rather than not started?

Reply:
currently having it die upon exception raised -- download failures should be forgiven at the point of retry/timeout parameters

making subsequent calls to download() start new download threads would expose us to races and likely not speed up downloading